Pod/Classes/LSClockState.m

```diff
@@ -82,18 +86,20 @@ - (void)addSampleWithOriginMicros:(SInt64)originMicros
         latestOffsetMicros = ((receiveMicros - originMicros) + (transmitMicros - destinationMicros)) / 2;
     }
 
-    // Discard the oldest sample and push the new one.
-    [self.samples removeObjectAtIndex:0];
-    [self.samples
-        addObject:[[LSSyncSample alloc] initWithDelayMicros:latestDelayMicros offsetMicros:latestOffsetMicros]];
+    dispatch_async(self.samplesQueue, ^{
```

Comment:
I think you mean `dispatch_sync`? Assuming you are trying to use this queue for serialized access / as a memory barrier...